CONTRIBUTING.md

```diff
@@ -0,0 +1,107 @@
+# Contributing
+
+`cargo-semver-checks` queries `rustdoc`-generated crate documentation using the `trustfall`
+["query everything" engine](https://github.com/obi1kenobi/trustfall).
+Each query looks for a particular kind of semver violation, such as:
+- public struct was removed
+- public enum's variant was removed
+- public struct is no longer `Send/Sync/Debug/Clone` etc.
+- public enum has a new variant, but wasn't non-exhaustive in the last version
+
+## Design goals
```

Comment:
What do you think about moving the design goals to a separate file, maybe called `DESIGN.md` or similar? It feels a bit strange that the contributing doc starts with a long section that isn't really directly about contributing, and doesn't really address why that is the case.

Reply:
Maybe in the future?  DESIGN is a subset of CONTRIBUTING information so as we grow, it might become important but then we have to help people discover these files (for some use cases) so keeping things focused when smaller I think helps.